### ISSUE TITLE
navigateTo in middleware throws ERR_HTTP_HEADERS_SENT when redirecting to external URL on route without path

### ISSUE THREAD

**Issue by MikeBellika** [nuxt/nuxt#19557]
### Environment

- Operating System: `Darwin`
- Node Version:     `v18.12.1`
- Nuxt Version:     `3.2.3`
- Nitro Version:    `2.2.3`
- Package Manager:  `yarn@1.22.19`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

1. Clone and run https://github.com/MikeBellika/navigate-to-500 locally
2. `curl http://localhost:3000/notfound`
3. Look at logs

### Describe the bug

When using `navigateTo` in a middleware the following error shows up in the terminal:
```
[nitro] [dev] [unhandledRejection] Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at new NodeError (node:internal/errors:393:5)
    at ServerResponse.setHeader (node:_http_outgoing:644:11)
    at Object.errorhandler [as onError] (file:///Users/mike/code/playground/navigate-to-reproduction/.nuxt/dev/index.mjs:457:20)
    at Server.toNodeHandle (file:///Users/mike/code/playground/navigate-to-reproduction/node_modules/h3/dist/index.mjs:1278:27)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
```
This error only happens if:
1. The navigateTo points to an external site (with `external: true`)
2. The original path doesn't have a route

The behaviour is different in dev mode and in production:
* When running in dev mode, the error shows up every time in the console, however the redirect still happens. 
* When running in production on Vercel, the error is flaky and occurs on every ~1/3 requests. The redirect doesn't happen and the server returns a 500.

### Additional context

We're also seeing this error on paths that exists. This error is much more flaky and I haven't been able to reproduce it locally. From my testing it seems like it happens on approximately 1/1000 requests.

Feel free to move this issue upstream, if it's not a nuxt problem.

**Comment by ilya-buligin** [comment]
Has the same issue with redirects on my Vercel production :(

**Comment by ilya-buligin** [comment]
I found the temporal solution. It is a script that patches build files. Should be called after the build step on Vercel:
```
import fs from 'node:fs'
import path from 'node:path'

const original = fs.readFileSync(
  path.join(process.cwd(), '.vercel/output/functions/__nitro.func/chunks/nitro/vercel.mjs'),
  { encoding: 'utf8' },
)

console.log('Pathing...')

const newVercelFunction = `
const vercel = (function(req, res) {
  const query = req.headers["x-now-route-matches"];
  if (query) {
    console.log('query detected');
    const { url } = parseQuery(query);
    if (url) {
      req.url = url;
    }
  }

  function debounce(func, delay) {
    let timeoutId;
  
    return function (...args) {
      const context = this;
  
      clearTimeout(timeoutId);
  
      timeoutId = setTimeout(() => {
        func.apply(context, args);
      }, delay);
    }
  }

  const debouncedEnd = debounce(res.end, 10)
  res.end = debouncedEnd

  return handler(req, res);
})

`

const pathed = original.replace(/const vercel = \(.*\)/gs, newVercelFunction)

fs.writeFileSync(
  path.join(process.cwd(), '.vercel/output/functions/__nitro.func/chunks/nitro/vercel.mjs'),
  pathed,
)
```

**Comment by MikeBellika** [comment]
Some extra info: We switched to using routeRules for redirects on vercel, which fixed this bug. We can't redirect URLs with query params with this method but we don't have too many of those redirects.

Before we switched, we could see in the logs that we had a lot of `ERR_HTTP_HEADERS_SENT` errors in the logs. Multiple every minute and probably around 1-2% of all requests would give a 500 because of this. This happened even on pages without redirects. After removing the middleware this went to 0.
